modules/apps/object/object-test/src/testFunctional/macros/json/JSONObject.macro

```diff
@@ -483,6 +483,34 @@ definition {
 		JSONCurlUtil.post(${curl});
 	}
 
+	@summary = "This calls the JSON WS API to add a picklist with custom ERC"
+	macro addPicklistWithCustomERC {
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (!(isSet(userEmailAddress))) {
+			var userEmailAddress = "test@liferay.com";
+		}
+
+		if (!(isSet(userPassword))) {
+			var userPassword = "test";
+		}
+
+		var curl = '''
+			${portalURL}/o/headless-admin-list-type/v1.0/list-type-definitions \
+				-H 'Content-Type: application/json' \
+				-u ${userEmailAddress}:${userPassword} \
+				-d '{
+					"externalReferenceCode": "${externalReferenceCode}",
```

Comment:
What happens if we don't set an externalReferenceCode? @GABRIELCMSANTOS

Reply:
On this API, I was trying to resume creating the Picklist, changing the ERC, and adding items, as required by the previous commentary. And the custom ERC is needed for the automation.

Reply:
Ah let me rephrase the question. Can we use this macro to create the picklist without specifying the ERC? It's an unnecessary step for most tests so I wanted to know what happens if we call this macro without setting the ERC. If it works, that's fine. If it doesn't, then we should explore a better design so that we don't need to specify the ERC if we don't want to. Does that make sense?

Reply:
Yes, I got it. I will see if I can change the ERC after creating the picklist with the items.

Reply:
@GABRIELCMSANTOS Sorry for the delay on this, but I meant that you could have created the Picklist Items with API instead of the Picklist itself. Not entirely sure if this test even needs Picklist Items in the first place to be honest. 

I think JSONObject.addPicklist allows a user to set an ERC so this new macro you created might not be necessary. Test looks good though.

Action Items:
1. Test to see if JSONObject.addPicklist will allow you to use a custom ERC. If not, creating the Picklist by itself via UI is okay. JSONObject.addPicklistWithCustomERC shouldn't be needed, so you do not need to submit that, but thank you for your work on it. Sorry for the confusion. 
2. If needed, change the Picklist Items steps to API.

Reply:
@epark18 No problem at all about the delay!
About the creation of the items, I did that so it could be visually verified that the picklist was changed with the update. But if it's not necessary, I can remove it. And I will see if I will be able to use SONObject.addPicklist to generate the ERC. Thanks for clarifying!